File: LeetCode/Medium/0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.go
func findMin(nums []int) int {
    low := 0
    high := len(nums) - 1

    if nums[low] <= nums[high] {
        return nums[low]
    }

    mini := 9999

    for low <= high {
        mid := low + (high - low) / 2
        if nums[low] <= nums[mid] {
            mini = min(mini, nums[low])
            low = mid + 1
        } else {
            mini = min(mini, nums[mid])
            high = mid - 1
        }
    }

    return mini
}

File: LeetCode/Medium/0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        low = 0
        high = len(nums) - 1

        if nums[low] <= nums[high]:
            return nums[low]

        ans = 5001

        while (low <= high):
            if nums[low] <= nums[high]:
                ans = min(ans, nums[low])

            mid = (low + high) // 2

            if nums[low] <= nums[mid]:
                ans = min(ans, nums[low])
                low = mid + 1

            else:
                ans = min(ans, nums[mid])
                high = mid - 1

        return ans
        